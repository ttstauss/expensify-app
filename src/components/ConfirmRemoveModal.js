import React from 'react'
import Modal from 'react-modal'

const ConfirmRemoveModal = props => (
  <Modal
    isOpen={props.modalIsOpen}
    contentLabel="Remove?"
    closeTimeoutMS={200}
    className="modal"
  >
    <h3 className="modal__title">Are you sure?</h3>
    <p className="modal__subtitle">You will not be able to recover this expense.</p>
    <div className="button-group">
      <button className="button button--secondary" onClick={props.handleCancelRemove}>Cancel</button>
      <button className="button button--warning" onClick={props.handleConfirmRemove}>Yes, delete it!</button>
    </div>
  </Modal>
)

if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app')

export default ConfirmRemoveModal