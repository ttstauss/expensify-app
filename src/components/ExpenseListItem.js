import React from 'react'
import { Link } from 'react-router-dom'

export default ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{amount} - {createdAt}</p>
  </div>
)