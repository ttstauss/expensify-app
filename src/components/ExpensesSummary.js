import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = ({ expenseCount, expensesTotal, hiddenExpenses }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
        {hiddenExpenses > 0 && <p className="page-header__message">Not showing {hiddenExpenses} expenses</p>}
      </div>
    </div>
  )
}

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  const hiddenExpenses = state.expenses.length - visibleExpenses.length
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses),
    hiddenExpenses: hiddenExpenses
  }
}

export default connect(mapStateToProps)(ExpensesSummary)