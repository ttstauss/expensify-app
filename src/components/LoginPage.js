import React from 'react'
import { connect } from 'react-redux'
import { startGoogleLogin, startFacebookLogin } from '../actions/auth'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faGoogle, faFacebookF } from '@fortawesome/free-brands-svg-icons'

export const LoginPage = ({ startGoogleLogin, startFacebookLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>It's time to get your expenses under control</p>
      <button className="button button--google" onClick={startGoogleLogin}>
        <FontAwesomeIcon className="social-icon" icon={faGoogle} />
        Login with Google
      </button>
      <button className="button button--facebook button--last" onClick={startFacebookLogin}>
        <FontAwesomeIcon className="social-icon" icon={faFacebookF} />
        Login with Facebook
      </button>
    </div>
  </div>
)

const mapDispatchToProps = dispatch => ({
  startGoogleLogin: () => dispatch(startGoogleLogin()),
  startFacebookLogin: () => dispatch(startFacebookLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)