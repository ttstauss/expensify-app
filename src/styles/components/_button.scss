.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}

.button--warning {
  background: #c62828;
}

.button-group {
  display: flex;
  justify-content: space-between;
}

%button--social-shared {
  margin: 0 0 $s-size 0;
  transition: background .3s ease;
  width: 100%;
}

.button--google {
  @extend %button--social-shared;
  background: #db3236;
  &:hover {
    background: lighten(#db3236, 7%)
  }
}

.button--facebook {
  @extend %button--social-shared;
  background: #3B5998;
  &:hover {
    background: lighten(#3B5998, 7%)
  }
}

.button--last {
  margin: 0;
}