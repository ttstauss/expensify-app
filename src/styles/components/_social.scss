.social-icon {
  margin: 0 $s-size 0 0;
}