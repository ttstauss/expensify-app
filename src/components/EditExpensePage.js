import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import ConfirmRemoveModal from '../components/ConfirmRemoveModal'

export class EditExpensePage extends React.Component {
  state = {
    modalIsOpen: false,
    confirmRemove: false
  }
  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }
  openModal = () => {
    this.setState(() => ({ modalIsOpen: true }))
  }
  handleConfirmRemove = () => {
    this.setState(() => ({
      modalIsOpen: false,
      confirmRemove: true
    }))
    this.props.startRemoveExpense({ id: this.props.expense.id })
    this.props.history.push('/')
  }
  handleCancelRemove = () => {
    this.setState(() => ({ modalIsOpen: false }))
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <div className="content-body">
            <ExpenseForm
              expense={this.props.expense}
              onSubmit={this.onSubmit}
            />
            <div>
              <button className="button button--secondary" onClick={this.openModal}>Remove Expense</button>
            </div>
          </div>
        </div>
        <ConfirmRemoveModal
          modalIsOpen={this.state.modalIsOpen}
          handleCancelRemove={this.handleCancelRemove}
          handleConfirmRemove={this.handleConfirmRemove}
        />
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)