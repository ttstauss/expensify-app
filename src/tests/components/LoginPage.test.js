import React from 'react'
import { shallow } from 'enzyme'
import { LoginPage } from '../../components/LoginPage'

test('should render LoginPage correclty', () => {
  const wrapper = shallow(<LoginPage />)
  expect(wrapper).toMatchSnapshot()
})

test('should call startGoogleLogin on button click', () => {
  const startGoogleLogin = jest.fn()
  const wrapper = shallow(<LoginPage startGoogleLogin={startGoogleLogin} />)
  wrapper.find('.button--google').simulate('click')
  expect(startGoogleLogin).toHaveBeenCalled()
})

test('should call startFacebookLogin on button click', () => {
  const startFacebookLogin = jest.fn()
  const wrapper = shallow(<LoginPage startFacebookLogin={startFacebookLogin} />)
  wrapper.find('.button--facebook').simulate('click')
  expect(startFacebookLogin).toHaveBeenCalled()
})