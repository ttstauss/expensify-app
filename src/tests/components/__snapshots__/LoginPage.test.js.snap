// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correclty 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control
    </p>
    <button
      className="button button--google"
    >
      <FontAwesomeIcon
        border={false}
        className="social-icon"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              488,
              512,
              Array [],
              "f1a0",
              "M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z",
            ],
            "iconName": "google",
            "prefix": "fab",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
      Login with Google
    </button>
    <button
      className="button button--facebook button--last"
    >
      <FontAwesomeIcon
        border={false}
        className="social-icon"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              264,
              512,
              Array [],
              "f39e",
              "M76.7 512V283H0v-91h76.7v-71.7C76.7 42.4 124.3 0 193.8 0c33.3 0 61.9 2.5 70.2 3.6V85h-48.2c-37.8 0-45.1 18-45.1 44.3V192H256l-11.7 91h-73.6v229",
            ],
            "iconName": "facebook-f",
            "prefix": "fab",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
      Login with Facebook
    </button>
  </div>
</div>
`;
